File: Fundamentos da Programação Orientada a Objetos/Exercicios_Manipulacao_Arquivo_p3-main/ex03/entidade/Produto.java
package entidade;

public class Produto {
	 private int codigo;
	    private int quantidade;
	    private String nome;
	    private String tamanho;
	    private double valor;

	    public Produto(int codigo, int quantidade, String nome, String tamanho, double valor) {
	        this.codigo = codigo;
	        this.quantidade = quantidade;
	        this.nome = nome;
	        this.tamanho = tamanho;
	        this.valor = valor;
	    }

	    @Override
	    public String toString() {
	        return "Produto[" +
	                " Codigo: " + codigo +
	                ", Quantidade: " + quantidade +
	                ", Nome: " + nome + 
	                ", Tamanho: " + tamanho + 
	                ", Valor: " + valor +
	                " ]";
	    }
}

File: Programação Web Front-End/Lista_Array_POO/exerc_01/js/Produto.js
export default class Produto {
    #nome;
    #codigo;
    #quantidadeEmEstoque;
    #dataValidade;
    
    constructor(nome, codigo, dataValidade, quantidadeEmEstoque) {
        this.#nome = nome;
        this.#codigo = codigo;
        this.#dataValidade = dataValidade;
        this.#quantidadeEmEstoque = quantidadeEmEstoque;
    }

    get getNome() {
        return this.#nome;
    }

    set setNome(nome) {
        this.#nome = nome;
    }

    get getCodigo() {
        return this.#codigo;
    }

    set setCodigo(codigo) {
        this.#codigo = codigo;
    }

    get getDataValidade() {
        return this.#dataValidade;
    }

    set setDataValidade(dataValidade) {
        this.#dataValidade = dataValidade;
    }

    get getQuantidadeEmEstoque() {
        return this.#quantidadeEmEstoque;
    }

    set setQuantidadeEmEstoque(quantidadeEmEstoque){
        this.#quantidadeEmEstoque = quantidadeEmEstoque;
    }
    
    exibirConteudo() {
        console.log(`Nome: ${this.#nome}`);
        console.log(`Código: ${this.#codigo}`);
        console.log(`Quantidade em Estoque: ${this.#quantidadeEmEstoque}`);
        console.log(`Data de Validade: ${this.#dataValidade}`);
    }
}


